### PR TITLE
feat(rawdata): create rawdata features

### PR DESCRIPTION
**Pull request by hmartiins**
### **Proposed changes**

creating rawdata features, this includes your repositories, services, controllers, routes ..

<br>

### CODE REVIEW

File: src/modules/rawdata/dtos/ICreateInterestPointDTO.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export default interface ICreateInterestPointDTO {
```
Comment (hmartiins) [inline src/modules/rawdata/dtos/ICreateInterestPointDTO.ts]: the DTO should be called
```ts
IInterestPointDTO
```

File: src/modules/rawdata/dtos/ICreateInterestPointTypeDTO.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export default interface ICreateInterestPointTypeDTO {
```
Comment (hmartiins) [inline src/modules/rawdata/dtos/ICreateInterestPointTypeDTO.ts]: the DTO should be called
```ts
IInterestPointTypeDTO
```

File: src/modules/rawdata/infra/http/controllers/InterestPointController.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+    const interestPoint = await showInterestPoint.execute({
+      interest_point_id,
+    });
+
+    return response.status(200).json({ interestPoint });
+  }
+
+  public async index(request: Request, response: Response): Promise<Response> {
+    const interest_point_id: string = (request.params as unknown) as string;
```
Comment (hmartiins) [inline src/modules/rawdata/infra/http/controllers/InterestPointController.ts]: there is no need to pass an id in the index method parameter, it has no use

File: src/modules/rawdata/infra/http/controllers/InterestPointTypeController.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    const interestPoint = await createInterestPointType.execute({
+      name,
+      description,
+    });
+
+    return response.json(interestPoint);
+  }
+
+  public async index(request: Request, response: Response): Promise<Response> {
+    const interest_point_type_id: string = (request.params as unknown) as string;
```
Comment (hmartiins) [inline src/modules/rawdata/infra/http/controllers/InterestPointTypeController.ts]: there is no need to pass an id in the index method parameter, it has no use

File: src/modules/rawdata/repositories/IInterestPointRepository.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import InterestPoint from '../infra/typeorm/entities/InterestPoint';
+import ICreateInterestPointDTO from '../dtos/ICreateInterestPointDTO';
+
+export default interface IInterestPointRepository {
+  findAllInterestPoint(
```
Comment (hmartiins) [inline src/modules/rawdata/repositories/IInterestPointRepository.ts]: there is no need to pass an id in the index method parameter, it has no use

File: src/modules/rawdata/repositories/IInterestPointTypeRepository.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import InterestPointType from '../infra/typeorm/entities/InterestPointType';
+import ICreateInterestPointTypeDTO from '../dtos/ICreateInterestPointTypeDTO';
+
+export default interface IInterestPointTypeRepository {
+  findAllInterestPointType(
+    expect_interest_point_type_id?: string,
```
Comment (hmartiins) [inline src/modules/rawdata/repositories/IInterestPointTypeRepository.ts]: there is no need to pass an id in the index method parameter, it has no use

File: src/modules/rawdata/services/interestPoint/CreateInterestPointService.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    number,
+    district,
+    interest_point_type_id,
+  }: ICreateInterestPointDTO): Promise<InterestPoint> {
+    const checkInterestPointExists = await this.interestPointRepository.findByName(
+      name,
+    );
+
+    if (checkInterestPointExists) {
+      throw new AppError('Name already used.');
```
Comment (hmartiins) [inline src/modules/rawdata/services/interestPoint/CreateInterestPointService.ts]: put a 409 error in that error

File: src/modules/rawdata/services/interestPoint/ListInterestPointService.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
+import { injectable, inject } from 'tsyringe';
```
Comment (hmartiins) [inline src/modules/rawdata/services/interestPoint/ListInterestPointService.ts]: change the file name to IndexInterestPointService, to maintain verbal standards

File: src/modules/rawdata/services/interestPointType/CreateInterestPointTypeService.ts [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+  public async execute({
+    name,
+    description,
+  }: IRequest): Promise<InterestPointType> {
+    const checkNameExists = await this.interestPointTypeRepository.findByName(
+      name,
+    );
+    if (checkNameExists) {
+      throw new AppError('Interest point type already exists', 403);
```
Comment (hmartiins) [inline src/modules/rawdata/services/interestPointType/CreateInterestPointTypeService.ts]: the correct http code would be 409 and not 403

File: src/modules/rawdata/services/touristSpot/CreateTouristSpotService.ts [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    description,
+    latitude,
+    longitude,
+  }: ITouristSpotDTO): Promise<TouristSpot> {
+    const checkTouristSpotExists = await this.touristSpotRepository.findByName(
+      name,
+    );
+
+    if (checkTouristSpotExists) {
+      throw new AppError('Name already used.');
```
Comment (hmartiins) [inline src/modules/rawdata/services/touristSpot/CreateTouristSpotService.ts]: enter the code http 409

File: src/shared/infra/azure/imageStorage/imageDelete.ts [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+): Promise<boolean> {
+  const azureKey = process.env.AZURE_STORAGE_KEY || '';
+  const image = imageURL.split('images/');
+
+  const blobServiceClient = BlobServiceClient.fromConnectionString(azureKey);
+  const containerClient = blobServiceClient.getContainerClient(blobContainer);
+  const blockBlobClient = containerClient.getBlockBlobClient(image[1]);
+
+  await blockBlobClient.download(0);
+  blockBlobClient.delete();
```
Comment (hmartiins) [inline src/shared/infra/azure/imageStorage/imageDelete.ts]: throw an error in the code if the code block above fails

**Review by hmartiins — COMMENTED** [review]
I made some comments on parts of the code, check if I requested any changes in your development part

@devLeSilverio @joaovds @PocketLocked @dio99o

**Comment by hmartiins** [comment]
If any taxpayer wants to contest my verification or even do the verification, don't be afraid, just do it and interact here !!

File: src/modules/rawdata/services/interestPoint/ListInterestPointService.ts (hunk 8 above)
Comment (devLeSilverio) [reply src/modules/rawdata/services/interestPoint/ListInterestPointService.ts]: I changed the name both here and in the "ListInterestPointTypeService", thanks

File: src/modules/rawdata/dtos/ICreateInterestPointDTO.ts (hunk 1 above)
Comment (PocketLocked) [reply src/modules/rawdata/dtos/ICreateInterestPointDTO.ts]: Done!

File: src/modules/rawdata/dtos/ICreateInterestPointTypeDTO.ts (hunk 2 above)
Comment (PocketLocked) [reply src/modules/rawdata/dtos/ICreateInterestPointTypeDTO.ts]: Done!

File: src/modules/rawdata/services/interestPoint/CreateInterestPointService.ts (hunk 7 above)
Comment (PocketLocked) [reply src/modules/rawdata/services/interestPoint/CreateInterestPointService.ts]: Done!

File: src/modules/rawdata/services/interestPointType/CreateInterestPointTypeService.ts (hunk 9 above)
Comment (PocketLocked) [reply src/modules/rawdata/services/interestPointType/CreateInterestPointTypeService.ts]: Done!

File: src/modules/rawdata/services/touristSpot/CreateTouristSpotService.ts (hunk 10 above)
Comment (PocketLocked) [reply src/modules/rawdata/services/touristSpot/CreateTouristSpotService.ts]: Done!

**Review by hmartiins — COMMENTED** [review]
LGTM !

**Pull request merged**